integration_test/third_party_apps_test.go

```diff
@@ -432,22 +385,36 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 	return nonRetryable, nil
 }
 
-// Returns the authoritative set of all apps available for testing.
-// The authoritative list corresponds to the directory names under
-// integration_test/third_party_apps_data/applications
-func determineAllApps(t *testing.T) map[string]bool {
-	allApps := make(map[string]bool)
+// Returns a map of application name to its parsed and validated metadata.yaml.
+// The set of applications returned is authoritative and corresponds to the
+// directory names under integration_test/third_party_apps_data/applications.
+func fetchAppsAndMetadata(t *testing.T) map[string]common.IntegrationMetadata {
+	allApps := make(map[string]common.IntegrationMetadata)
 
-	files, err := os.ReadDir("third_party_apps_data/applications")
+	files, err := scriptsDir.ReadDir(path.Join("third_party_apps_data", "applications"))
 	if err != nil {
 		t.Fatalf("got error listing files under third_party_apps_data/applications: %v", err)
 	}
 	for _, file := range files {
-		if file.IsDir() {
-			allApps[file.Name()] = true
+		app := file.Name()
+		var metadata common.IntegrationMetadata
+		testCaseBytes, err := readFileFromScriptsDir(path.Join("applications", app, "metadata.yaml"))
+		if err != nil {
+			t.Fatalf("could not read applications/%v/metadata.yaml: %v", app, err)
 		}
+		err = yaml.UnmarshalStrict(testCaseBytes, &metadata)
+		if err != nil {
+			t.Fatalf("could not unmarshal contents of applications/%v/metadata.yaml: %v", app, err)
+		}
+		if err = validate.Struct(&metadata); err != nil {
+			t.Fatalf("could not validate contents of applications/%v/metadata.yaml: %v", app, err)
+		}
+		allApps[app] = metadata
+	}
+	log.Printf("found %v apps", len(allApps))
```

Comment:
The impacted apps list is still logged, on line 460.

I think it's fine to just log the number of apps here, since at this point there's no realistic way for it to be wrong anyway. It has to have a valid `metadata.yaml` file for each app or it will return an error and never get to this point anyway.